rules/aws_iam_policy_invalid_length.go

```diff
@@ -0,0 +1,61 @@
+package rules
+
+import (
+	"fmt"
+	hcl "github.com/hashicorp/hcl/v2"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-aws/project"
+)
+
+// AwsIAMPolicyInvalidLengthRule checks that the policy length is less than 2048 characters
+type AwsIAMPolicyInvalidLengthRule struct {
+	resourceType  string
+	attributeName string
+}
+
+// NewAwsIAMPolicyInvalidLengthRule returns new rule with default attributes
+func NewAwsIAMPolicyInvalidLengthRule() *AwsIAMPolicyInvalidLengthRule {
+	return &AwsIAMPolicyInvalidLengthRule{
+		resourceType:  "aws_iam_policy",
+		attributeName: "policy",
+	}
+}
+
+// Name returns the rule name
+func (r *AwsIAMPolicyInvalidLengthRule) Name() string {
+	return "aws_iam_policy_invalid_length"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *AwsIAMPolicyInvalidLengthRule) Enabled() bool {
+	return true
+}
+
+// Severity returns the rule severity
+func (r *AwsIAMPolicyInvalidLengthRule) Severity() string {
+	return tflint.ERROR
+}
+
+// Link returns the rule reference link
+func (r *AwsIAMPolicyInvalidLengthRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check checks the length of the policy
+func (r *AwsIAMPolicyInvalidLengthRule) Check(runner tflint.Runner) error {
+	return runner.WalkResourceAttributes(r.resourceType, r.attributeName, func(attribute *hcl.Attribute) error {
+		var policy string
+		err := runner.EvaluateExpr(attribute.Expr, &policy, nil)
+
+		return runner.EnsureNoError(err, func() error {
+			if len(policy) > 2048  {
+				runner.EmitIssueOnExpr(
+					r,
+					fmt.Sprintf("The policy length is %d characters and is limited to 2048 characters.", len(policy)),
```

Comment:
2048 characters is a limit for inline policies. The managed policy limit is 6144 characters and should not count white spaces.
https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html#reference_iam-quotas-entity-length

Reply:
Ahh. Yeah, that makes sense. Given it's entity based, and we would then also have to look at what it's being attached to we may not be able to apply a strict check to this except the highest option which is actually 10,240 (inline role policy). 

However, we could look at doing this for specifically as
aws_iam_group_policy: 5120
aws_iam_role_policy: 10240
aws_iam_user_policy: 2048

Unless, you know a way we can find what entity the policy is going to be attached to I can't think of any other option.

Reply:
Hmm, since `aws_iam_policy` always creates a [customer managed policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-vs-inline.html#customer-managed-policies), isn't the upper limit 6,144?
I think the upper limit for `inline_policy` blocks such as `aws_iam_role` resources will be 10,240.
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role#example-of-exclusive-inline-policies

Reply:
@wata727 Hmm. yeah, That makes more sense... I think the wording in this document kept throwing me off: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_grammar.html#policies-grammar-notes

Reading the limits page again. I think you are right aws_iam_policy should be good to set to 6144. And I am not sure the inlines can be checked given the docs " But the total aggregate policy size (the sum size of all inline policies) per entity cannot exceed the following quotas"

My apologies on the confusion here.